build.gradle

```diff
@@ -134,6 +136,72 @@ spotless {
 		trimTrailingWhitespace();
 	}
 }
+
+task writeGitInfo {
+	def hasGit = { ->
+		try {
+			exec {
+				executable 'git'
+				delegate.args '--version'
+				standardOutput = OutputStream.nullOutputStream()
+			}
+		} catch (Exception e) {
+			return false
+		}
+		return true
+	}
+
+	def addGitOutput = { PrintStream output, String name, String... args ->
+		output.println("### $name (git ${String.join(' ', args)})")
+		// Temporary stream so that we don't close output
+		def execOutput = new ByteArrayOutputStream()
+		def ex = exec {
+			workingDir projectDir
+			executable 'git'
+			delegate.args args
+			standardOutput = execOutput
+		}
+		output.print(execOutput)
+	}
+
+	def convertPath = { String path ->
+		String username = System.getenv("USERNAME")
```

Comment:
(Minor as most of us use Windows) Can we switch this to `System.getProperty("user.name")` so we can support OSes that don't set the `USERNAME` environment variable? (I got `null` when testing in Ubuntu).

[Reference](https://stackoverflow.com/questions/38686481/how-to-get-userid-and-gid-in-gradle) (Ironically, I got the information needed from the question, and not the answer).

Reply:
I decided to leave both in so that if possible we could get something that could be pasted into a shell (assuming it's the same user).